flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/factories/PrintTableSinkFactory.java

```diff
@@ -95,19 +96,22 @@ public DynamicTableSink createDynamicTableSink(Context context) {
 		return new PrintSink(
 				context.getCatalogTable().getSchema().toPhysicalRowDataType(),
 				options.get(PRINT_IDENTIFIER),
-				options.get(STANDARD_ERROR));
+				options.get(STANDARD_ERROR),
+				options.getOptional(FactoryUtil.SINK_PARALLELISM).orElse(null));
 	}
 
 	private static class PrintSink implements DynamicTableSink {
 
 		private final DataType type;
 		private final String printIdentifier;
 		private final boolean stdErr;
+		private final Integer parallelism;
```

Comment:
```suggestion
		private final @Nullable Integer parallelism;
```

Would be better to add `@Nullable` annotation. 

See the Flink Code Style Guideline: https://flink.apache.org/contributing/code-style-and-quality-java.html#java-optional